sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs

```diff
@@ -1091,43 +1108,148 @@ private void SetNameFieldsIfNull()
             bool rangeGetContentHash = default,
             CancellationToken cancellationToken = default) =>
             await DownloadStreamingInternal(
-                range,
-                conditions,
-                rangeGetContentHash,
-                $"{nameof(BlobBaseClient)}.{nameof(DownloadStreaming)}",
+                new BlobBaseDownloadOptions
+                {
+                    Range = range,
+                    Conditions = conditions,
+                    TransactionalHashingOptions = rangeGetContentHash
+                        ? new DownloadTransactionalHashingOptions()
+                        {
+                            Algorithm = TransactionalHashAlgorithm.MD5
+                        }
+                        : default,
+                    OperationName = $"{nameof(BlobBaseClient)}.{nameof(DownloadStreaming)}"
+                },
                 true, // async
                 cancellationToken)
                 .ConfigureAwait(false);
 
+        /// <summary>
+        /// The <see cref="DownloadStreaming(BlobBaseDownloadOptions, CancellationToken)"/>
+        /// operation downloads a blob from the service, including its metadata
+        /// and properties.
+        ///
+        /// For more information, see
+        /// <see href="https://docs.microsoft.com/rest/api/storageservices/get-blob">
+        /// Get Blob</see>.
+        /// </summary>
+        /// <param name="options">
+        /// Optional parameters.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response{BlobDownloadStreamingResult}"/> describing the
+        /// downloaded blob.  <see cref="BlobDownloadStreamingResult.Content"/> contains
+        /// the blob's data.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        ///
+        /// This API gives access directly to network stream that should be disposed after usage.
+        /// Consider the following alternatives:
+        /// <list type="bullet">
+        ///     <item>
+        ///         <term>DownloadContentAsync</term>
+        ///         <description>as a prefered way of downloading small blobs that can fit into memory</description>
+        ///     </item>
+        ///     <item>
+        ///         <term>DownloadToAsync</term>
+        ///         <description>to stream blob content to a path or a <see cref="Stream"/></description>
+        ///     </item>
+        /// </list>
+        /// </remarks>
+        [ForwardsClientCalls]
```

Comment:
We might be able to remove this annotation if we create a new DiagnosticScope in this method.

Reply:
It looks as if we can just remove these regardless. They must have been leftover from some struggles I was having.